core/src/test/java/io/aiven/klaw/service/TopicControllerServiceTest.java

```diff
@@ -382,74 +685,219 @@ public void getAllTopics() {
         .thenReturn(getSyncTopics("topic", 4));
     when(manageDatabase.getKafkaEnvList(anyInt())).thenReturn(utilMethods.getEnvLists());
     when(manageDatabase.getTeamNameFromTeamId(anyInt(), anyInt()))
-        .thenReturn("INFRATEAM", "INFRATEAM", "INFRATEAM", "INFRATEAM");
+        .thenReturn(
+            KwConstants.INFRATEAM,
+            KwConstants.INFRATEAM,
+            KwConstants.INFRATEAM,
+            KwConstants.INFRATEAM);
     when(mailService.getEnvProperty(anyInt(), anyString())).thenReturn("1");
+    when(commonUtilsService.groupTopicsByEnv(any())).thenReturn(getSyncTopics("topic", 4));
 
     List<List<TopicInfo>> topicsList =
         topicControllerService.getTopics(envSel, pageNo, "", topicNameSearch, null, null);
 
     assertThat(topicsList).hasSize(2);
   }
 
-  // topicSearch does not exist in topic names
   @Test
-  @Order(18)
-  public void getTopicsSearchFailure() throws Exception {
-    String envSel = "1", pageNo = "1", topicNameSearch = "demo";
+  @Order(36)
+  public void getTopicsSuccess2() {
+    String envSel = "1", pageNo = "1", topicNameSearch = "top";
 
     stubUserInfo();
-    when(manageDatabase.getTeamsAndAllowedEnvs(anyInt(), anyInt()))
-        .thenReturn(Collections.singletonList("1"));
+    when(commonUtilsService.getEnvsFromUserId(anyString()))
+        .thenReturn(new HashSet<>(Collections.singletonList("1")));
     when(commonUtilsService.deriveCurrentPage(anyString(), anyString(), anyInt())).thenReturn("1");
-    when(handleDbRequests.getSyncTopics(envSel, null, 1)).thenReturn(getSyncTopics("topic", 4));
+    when(handleDbRequests.getSyncTopics(any(), any(), anyInt()))
+        .thenReturn(getSyncTopics("topic", 4));
+    when(manageDatabase.getKafkaEnvList(anyInt())).thenReturn(utilMethods.getEnvLists());
+    when(manageDatabase.getTeamNameFromTeamId(anyInt(), anyInt()))
+        .thenReturn(
+            KwConstants.INFRATEAM,
+            KwConstants.INFRATEAM,
+            KwConstants.INFRATEAM,
+            KwConstants.INFRATEAM);
+    when(handleDbRequests.selectAllTopicsByTopictypeAndTeamname(anyString(), anyInt(), anyInt()))
+        .thenReturn(getSyncTopics("topic", 4));
+    when(mailService.getEnvProperty(anyInt(), anyString())).thenReturn("1");
+    when(commonUtilsService.groupTopicsByEnv(any())).thenReturn(getSyncTopics("topic", 4));
+    when(commonUtilsService.getFilteredTopicsForTenant(any())).thenReturn(utilMethods.getTopics());
 
     List<List<TopicInfo>> topicsList =
-        topicControllerService.getTopics(envSel, pageNo, "", topicNameSearch, null, null);
+        topicControllerService.getTopics(
+            envSel, pageNo, "", topicNameSearch, KwConstants.INFRATEAM, AclType.PRODUCER.value);
 
-    assertThat(topicsList).isNull();
+    assertThat(topicsList).isNull(); // for this filter criteria, there are no topics.
   }
 
   @Test
-  @Order(19)
-  public void getSyncTopics() throws Exception {
+  @Order(37)
+  public void getTopicsSuccess3() {
```

Comment:
We should give it a proper name instead of `1` and `2`.

Reply:
Agree, updated.